Ported over work from https://github.com/robbynshaw/textile-js-sdk

robbynshaw opened the pull request:
Utilizing the example from https://github.com/textileio/minimal-client-demo/blob/master/src/js/index.js, I created an example folder with a dropzone example in it, mirroring it almost exactly. The extent of the client functionality thus far is essentially shown in that example.

I separated out the milling stuff into it's own class of static methods for doing the *pre-processing* work, and just call that from within the threads module itself. (Not because I thought @carsonfarmer 's work was inadequate, but for my own edification and to document the process somewhat better in code. Coming from the outside, the milling stuff is a little hard to grasp.) My assumption is that the user of the library doesn't really need to be aware of how all of that works--they should just add a file and go.

I haven't tried to add a file with a stream yet, but I added a method for it. Axios is supposed to work fine with that, but the real test case will be in the Flickr export/import tool. If I find it needs changes based on that, I will post them up here as well.

Also, I didn't set any of this up with Yarn, as I don't have any experience with that and wanted to get it up here as soon as possible.

robbynshaw committed:
Ported over work from https://github.com/robbynshaw/textile-js-sdk

In browser file drop works with the client currently.

carsonfarmer commented on core/connection.js:
```diff
@@ -0,0 +1,26 @@
+    const opts = Connection.cleanOpts(options);
+
+    return axios.create({
+      baseURL: `${opts.url}:${opts.port}` // TODO this is flaky
```
This assumes it will always have a port option, which it might not if the API is exposed via public URL or similar. Might want to conditionally include the `:` when a port is specified.

robbynshaw replied:
I added a fix for this, but the cleanOpts method is still defaulting it to 40601 if it's empty, which means that one would have to pass in 80 or 443 in the above situation. I'm not sure here if you are looking to set the default to 80/empty or not. Feel free to change it around, just might need to change the usage in the README then as well.

carsonfarmer commented on docs/textile-js-sdk/0.1.0/Connection.html:
```diff
@@ -0,0 +1,258 @@
+<!DOCTYPE html>
```
Should probs change the path names here to reflect the 'new' name: `docs/js-textile-http-client/0.1.0/...`?

carsonfarmer replied:
I'm assuming these are auto-generated, but couldn't find a better place to make that comment 🤓

carsonfarmer reviewed (approved):
Wow @robbynshaw this is looking awesome. Really like the modular design here! There are a few minor nits in my comments, which you can fix now, or later, at your discretion. Might be nice to just merge this and then you can start working off the official version. Let's also consider setting things up to release to npm etc. Not a requirement though!

carsonfarmer commented:
I left it a bit ambiguous, but if you're happy to have this merged and work from there, just let me know.

carsonfarmer commented:
One more request here, could you update the README to include information about installing, testing, linting, etc?

robbynshaw committed:
Added usage to the README. Deleted and rebuilt docs with new name in package.json.

Renamed all the modules to better match the naming conventions in textile-go.
Fixed the url for better resiliency.

robbynshaw commented:
In addition to the above, I also renamed the modules to be 'threads' instead of 'thread' for example, to better match the naming in textile-go. Feel free to tell me to undo that or undo it yourself if you want, it was just bugging me, and I felt like this might be my last chance for a breaking change like that. :)

In retrospect, I should have broken this out into more commits. The github page is gonna look crazy for a while. Sorry.

One last note, I didn't add anything to the documentation section in the README as I was not sure how you wanted to handle that. You can point the github pages at the docs if you want, or do whatever you want. The docs are generated simply with jsdoc, which must just take the naming in package.json, so destroying the folder and rebuilding worked perfectly.

carsonfarmer reviewed (approved):
Let's do this!
